### DISCUSSION TITLE
Parallel on GPU

### DISCUSSION THREAD

**Discussion by EnxEng** [GeoStat-Framework/GSTools#186] [Ideas]
Hi to all,
great code gstools.
I was working more by generating multiple fields, by changing correlation length and variance, in 2d or 3d and I noted that the process could be speed up by using CUDA.

Have also a look here:
https://wp.unil.ch/geocomputing/software/grfs/
https://wp.unil.ch/geocomputing/files/2019/07/R%C3%A4ss_etal_2019_random3D.pdf

A good idea should be to incorporate a new module with and option to process random fields with gpus.

All the best!!!

**Comment by MuellerSeb** [comment]
Hey there!
Thanks for your interest in GSTools.

That paper looks interesting. This is the exact same way, we calculate our random fields with the Randomization method as documented here:
https://doi.org/10.1016/j.envsoft.2014.01.013

We already use parallel calculations to generate the fields:
https://github.com/GeoStat-Framework/GSTools/blob/57b94b1f488910b29f6d959fc29fa668c388be2a/gstools/field/summator.pyx#L35

You can compile GSTools with [OpenMP support](https://geostat-framework.readthedocs.io/projects/gstools/en/stable/index.html#pip). In the future there might also be GPU support directly from Cython:
https://github.com/cython/cython/pull/3449

For parallel sampling we also thought about using `schwimmbad` with `emcee`: https://github.com/GeoStat-Framework/GSTools/issues/35

What are your exact ideas on providing CUDA support?

Cheers,
Sebastian
